mindmeld/components/dialogue.py

```diff
@@ -1233,8 +1272,15 @@ def to_json(instance):
                 serialized_obj[attribute] = tuple(dict(item) for item in value)
             elif isinstance(value, immutables.Map):
                 serialized_obj[attribute] = dict(value)
+            elif value and isinstance(value, dict) and "entities" in value:
+                # Serialize slot-filling form
+                if value["entities"] and any(isinstance(i, FormEntity) for i in value["entities"]):
+                    value = dict(value)
```

Comment:
Why do we explicitly do `dict` here we already checked that `isinstance(value, dict)` in the above conditional?

Reply:
It fails to write into the dictionary without this explicit conversion